File: clients/algoliasearch-client-php/lib/Model/Insights/EventType.php
<?php

namespace Algolia\AlgoliaSearch\Model\Insights;

/**
 * EventType Class Doc Comment
 *
 * @category Class
 * @description An eventType can be a click, a conversion, or a view.
 *
 * @package Algolia\AlgoliaSearch
 */
class EventType
{
    /**
     * Possible values of this enum
     */
    const CLICK = 'click';

    const CONVERSION = 'conversion';

    const VIEW = 'view';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::CLICK,
            self::CONVERSION,
            self::VIEW,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/insights/EventType.java
package com.algolia.model.insights;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** An eventType can be a click, a conversion, or a view. */
@JsonAdapter(EventType.Adapter.class)
public enum EventType {
  CLICK("click"),

  CONVERSION("conversion"),

  VIEW("view");

  private final String value;

  EventType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static EventType fromValue(String value) {
    for (EventType b : EventType.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<EventType> {

    @Override
    public void write(final JsonWriter jsonWriter, final EventType enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public EventType read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return EventType.fromValue(value);
    }
  }
}
